src/handlers/musicHandler.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Responsible for creating and managing musicplayers per guild
+ */
+
+import { Collection } from 'discord.js';
+import MusicPlayer from '../models/MusicPlayer';
+import DiscordClient from '../models/client';
+
+export default class MusicHandler extends Collection<string, MusicPlayer> {
+    readonly client: DiscordClient;
+
+    constructor(client: DiscordClient) {
+        super();
+        this.client = client;
+    }
+
+    getOrCreate = (guildId: string): MusicPlayer =>
+        this.get(guildId) ?? this.create(guildId);
+
+    create = (guildId: string): MusicPlayer => {
```

Comment:
not sure if I like that pattern
that might make it less readable...

i like that create exists as a standalone function

Reply:
@Fire3galaxy Well, the issue is that MusicHandler is a Collection rather than a single instance of MusicPlayer. The idea is that there are potentially multiple MusicPlayer instances active, one per guild (discord server) so we need a way to manage multiple musicplayers

Hence, getorcreate(guildId) - if a guild (member) requests the bot play something, get the MusicPlayer instance for that server, or create a new one if it doesn't exist

Reply:
I'd be willing to set it to private perhaps